### ISSUE TITLE
Honor "external" options.

### ISSUE THREAD

**Issue by GordonSmith** [Swatinem/rollup-plugin-dts#70]
My package has dependencies on two third party libraries:
* @hpcc-js/common
* d3-drag

For my main bundle I have configured rollup to make "@hpcc-js/common" external (not bundled in my output), while "d3-drag" defaults to internal (so IS bundled inside my output).

When I look at the bundled "d" file the imports sections looks like this:
```
import { SVGWidget, TextBox, Widget, Text, Icon, SVGZoomWidget, SVGGlowFilter, ISize, PropertyExt, Selection, Utility, Palette, publish } from '@hpcc-js/common';
import { DragBehavior } from 'd3-drag';
```

I was expecting the `DragBehavior` to be inlined inside the bundled d.ts file?

Note: `DragBehavior` is a type which is getting auto resolved by tsc to `@types/d3-drag/index.d.ts`

**Comment by Swatinem** [comment]
Hm, I guess I can do that…

My reasoning is that for types, its better to dedupe these packages, same as with library code. But yes, its better to have a sane default and give authors the chance to override…

Give me a few days, I can’t really code right now…

**Comment by GordonSmith** [comment]
IMO this is a pretty big deal as the current situation (without rollup-plugin-dts) is that a packages NPM dependencies can get the "dependencies" and "devDependencies" sections correct for the bundled JS file, but it will be incorrect for the "type" files.

This meant (for the above example) that I was forced to move the "d3-drag" dependency from "devDependencies" (its correct place) into "dependencies", so end users would get the d3-drag types.

(which is a long winded way of saying thanks!  and that this has the potential to shrink a _lot_ of NPM dependencies out there)

**Comment by Swatinem** [comment]
Well its a bit unfortunate that `dependencies` (aka things that you need at *runtime* in *production*) and `devDependencies (things that you need to *develop* a library) does not match the usecase of developing when consuming a library.

I personally hate it when I see `@types` in the zipped archive that I deploy to production, so I personally prefer to manually add all the required `@types` into my local `devDependencies`, which sometimes is tedious and a huge pain, so I do get your concern.

**Comment by Swatinem** [comment]
Just published `1.2.0`, which exposes a new option.
However, `d3-drag` is a whooping ~400LOC alone, and pulls in `d3-selection` as well. So I would still recommend to just leave it as external instead of bundling…

**Comment by GordonSmith** [comment]
Thanks testing now.

Those 400LOC (its actually more OMM) is way better than having to include d3-drag in my dependencies (and if I omit it, my Angular users will have issues).

On the plus side looks like tree shaking is working as expected!

I have started to see some issues with unsupported syntax and the like - will open new issues as  I hit them...
